server/src/test/java/io/crate/planner/ExplainPlannerTest.java

```diff
@@ -174,13 +210,37 @@ public CompletableFuture<?> completionFuture() {
     @Test
     public void test_explain_on_collect_uses_cast_optimizer_for_query_symbol() throws Exception {
```

Comment:
I modified this test to use the `SimplifyEqualsOperationOnIdenticalReferences` rule instead of `SwapCastsInComparisonOperators`. I did this to show that the `CastOptimizer` is applied. Before this change, the `SwapCastsInComparisonOperators` rule was being applied, but its effects were not visible in the output generated by `ExplainPlan::printLogicalPlan`. In other words, removing the line in printLogicalPlan that applies the optimizer had no impact on the output.